contracts/HubPool.sol

```diff
@@ -638,6 +637,20 @@ contract HubPool is HubPoolInterface, Testable, Lockable, MultiCaller, Ownable {
         if (protocolFeesCaptured > 0) unclaimedAccumulatedProtocolFees[l1Token] += protocolFeesCaptured;
     }
 
-    // Added to enable the SpokePool to receive ETH. used when unwrapping WETH.
-    receive() external payable {}
+    // If _notEntered is false then this method has been called by the callback function by dropping ETH onto the
+    // contract. In this case, deposit the ETH into WETH. If true then this was set as a result of unwinding LP tokens,
+    // with the intention of sending ETH to the LP. In this case, do nothing as we intend on sending the ETH to the LP.
+    function depositEthToWeth() public payable {
+        if (_notEntered) weth.deposit{ value: address(this).balance }();
```

Comment:
As discussed IRL, let's expose `_notEntered` via a descriptive `public` function and call that here